src/app/auth/services/auth.service.ts

```diff
@@ -410,9 +420,101 @@ export const unblockAccount = (token: string): Promise<void> => {
   return authClient.unblockAccount(token);
 };
 
+export const signUp = async (
+  doSignUp: RegisterFunction | UpdateInfoFunction,
+  email: string,
+  password: string,
+  token: string,
+  isNewUser: boolean,
+  redeemCodeObject: boolean,
+  dispatch: AppDispatch,
+) => {
+  const { xUser, xToken, mnemonic } = isNewUser
+    ? await (doSignUp as RegisterFunction)(email, password, token)
+    : await (doSignUp as UpdateInfoFunction)(email, password);
+
+  localStorageService.clear();
+
+  localStorageService.set('xToken', xToken);
+  localStorageService.set('xMnemonic', mnemonic);
+
+  const xNewToken = await getNewToken();
+  localStorageService.set('xNewToken', xNewToken);
+
+  const privateKey = xUser.privateKey
+    ? Buffer.from(decryptPrivateKey(xUser.privateKey, password)).toString('base64')
+    : undefined;
+
+  const user = {
+    ...xUser,
+    privateKey,
+  } as UserSettings;
+
+  dispatch(userActions.setUser(user));
+  await dispatch(userThunks.initializeUserThunk());
+  dispatch(productsThunks.initializeThunk());
+
+  if (!redeemCodeObject) dispatch(planThunks.initializeThunk());
+  if (isNewUser) dispatch(referralsThunks.initializeThunk());
+  return { token: xToken, user: xUser, mnemonic };
+};
+
+export const logIn = async (
```

Comment:
Can you add tests to these functions?
They are quite sensitive to changes and it would be a good idea to have tests